code/modules/hydroponics/hydroponics.dm

```diff
@@ -212,6 +214,9 @@
 	. = ..()
 	if(!QDELETED(src) && gone == myseed)
 		set_seed(null, FALSE)
+	if(istype(gone, /obj/item/clothing/head/mob_holder/snail))
+		QDEL_NULL(our_snail)
```

Comment:
This looks wrong, why would you delete it when it exits the tray?
You probably want something more like

```suggestion
	if(gone == our_snail)
		our_snail = null
```